apps/src/templates/HideToolbarHelper.jsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import msg from '@cdo/locale';
+import cookies from 'js-cookie';
+import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
+import trackEvent from '../util/trackEvent';
+import _ from 'lodash';
+
+const styles = {
+  closeX: {
+    position: 'absolute',
+    top: 5,
+    right: 5,
+    fontSize: 13
+  }
+};
+
+// Note that additional styling can be found in apps/style/HideToolbarHelper.scss.
+
+const HidetoolbarHelperCookieName = 'hide_toolbar_helper';
+
+/**
+ * An overlay with instrutions on hiding the toolbar for iPhone with iOS 13.
+ */
+export default class HideToolbarHelper extends React.Component {
+  state = {
+    showHelper: false
+  };
+
+  constructor(props) {
+    super(props);
+
+    // Track whether we knew the toolbar was shown last update, so that
+    // if it disappears, we can track the event in analytics...
+    this.wasToolbarShowing = false;
+
+    // ...but only track it once.
+    this.trackedToolbarHide = false;
+  }
+
+  updateLayout = () => {
+    const isiOS13 = navigator.userAgent.indexOf('iPhone OS 13') !== -1;
+    const isHideCookieSet = cookies.get(HidetoolbarHelperCookieName);
+    const isLandscape = window.orientation !== 0;
+
+    // window.innerHeight is smaller than document.body.offsetHeight when
+    // the iOS 13 Safari toolbar is showing.  It becomes equal when the toolbar
+    // is hidden.  (Interestingly, document.documentElement.clientHeight is
+    // the same as document.body.offsetHeight in both cases.)
+    const isToolbarShowing = window.innerHeight < document.body.offsetHeight;
```

Comment:
[Not blocking] Do you know if this is coincidence or intentional? i.e. I'm a bit concerned that this may end up being fragile in the long-term.

Reply:
Oh completely agreed, which is one reason why I'm only doing this for iOS Safari 13, and not for 14 onward.  I did find one online post that had discovered the same: https://coderwall.com/p/yaro4w/fullscreen-experience-on-mobile-safari